independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/CreationalContextImpl.java

```diff
@@ -24,6 +24,9 @@ public class CreationalContextImpl<T> implements CreationalContext<T>, Function<
     private final CreationalContextImpl<?> parent;
     private List<InstanceHandle<?>> dependentInstances;
 
+    // may carry a current injection point or a current decorator delegate
+    private Object extra;
```

Comment:
I'm not quite sure if `extra` is a good name... maybe something along the lines of `currentObject` or so?

Reply:
I've ended up splitting it into 2: `currentInjectionPoint` and `currentDecoratorDelegate`.

Reply:
That's even better :+1: